JET update to enable fcst only run:

lgannoaa opened the pull request:
Changes to be committed:
	modified:   env/JET.env
	modified:   modulefiles/module_base.jet
	modified:   sorc/build_gfs_util.sh
	modified:   sorc/build_gldas.sh
	modified:   sorc/build_workflow_utils.sh
	modified:   sorc/link_fv3gfs.sh

lgannoaa committed:
JET update to enable fcst only run:
 Changes to be committed:
	modified:   env/JET.env
	modified:   modulefiles/module_base.jet
	modified:   sorc/build_gfs_util.sh
	modified:   sorc/build_gldas.sh
	modified:   sorc/build_workflow_utils.sh
	modified:   sorc/link_fv3gfs.sh

aerorahul commented on modulefiles/module_base.jet:
```diff
@@ -1,32 +1,29 @@
 
 module load gempak/7.4.2
 
+# Load from official NCEPLIBS
+module use -a /mnt/lfs4/HFIP/hfv3gfs/nwprod/NCEPLIBS/modulefiles
+module load prod_util/v1.0.18
+module load grib_util/v1.1.1
+module load g2tmpl/1.6.0
+module load crtm/2.3.0
+
+# Add new path for netcdfp  -- fix for netcdf HDF errors
+setenv NCEPLIBS  /lfs4/HFIP/hfv3gfs/gwv/l0701/lib/
+module use -a /mnt/lfs4/HFIP/hfv3gfs/gwv/l0701/lib/modulefiles
+module load esmflocal/8_0_1
+module load netcdfp/4.7.4
```
Jet is using hpc-stack for the model.  Please use those.  
See jet related files here:
https://github.com/ufs-community/ufs-weather-model/tree/develop/modulefiles

lgannoaa replied:
Modification made as requested

aerorahul commented on sorc/build_gldas.sh:
```diff
@@ -8,7 +8,11 @@ USE_PREINST_LIBS=${USE_PREINST_LIBS:-"true"}
 else
-  export MOD_PATH=${cwd}/lib/modulefiles
+  if [ $target = jet ]; then
+    export MOD_PATH=/lfs4/HFIP/hfv3gfs/nwprod/hpc-stack/libs/modulefiles/stack
```
using hpc-stack here, but not in `module_base.jet`.  Is there a reason for this inconsistency?

aerorahul replied:
Can you explain why this if-block is needed or used.

aerorahul replied:
@KateFriedman-NOAA Can you check why gldas needs this special treatment and others e.g. gsi, don't?

lgannoaa replied:
Modification mad as requested

aerorahul commented on sorc/build_workflow_utils.sh:
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -eux
 
+source ./machine-setup.sh > /dev/null 2>&1
```
This line is sourced on 14.  I presume you are looking to set the variable `target` here.  You don't need since the calling script `build_all.sh` sets this variable and passes it down.  See: https://github.com/lgannoaa/global-workflow/blob/b19ccea00d59a4c8ea13d9f8ffb00fbe7bfc9d49/sorc/build_all.sh#L154

lgannoaa replied:
Modification made as requested

aerorahul reviewed (changes requested):
See comments inline.

lgannoaa committed:
Include sdditional JET modulefiles and script

lgannoaa committed:
Update hpc-stack and clean up the build_workflow_utils.sh script
Changes to be committed:
	modified:   modulefiles/module_base.jet
	modified:   sorc/build_gldas.sh
	modified:   sorc/build_workflow_utils.sh

aerorahul commented on modulefiles/OznMonBuild.jet:
```diff
@@ -0,0 +1,17 @@
+# Modfule file for oznmon
+########################################################
+export ver=2.0.2
+export FCOMP=mpiifort
+
+export CF=$FCOMP
+export FC=$FCOMP
```
how can you set these before loading the compiler?

lgannoaa replied:
The JET build with this version of OznMonBuild.jet was working.

aerorahul replied:
It is working, but not because of this line.

Setting these variables here is doing nothing since the compiler module is overwriting it.

KateFriedman-NOAA replied:
We don't need the modulefiles/OznMonBuild.* and modulefiles/RadMonBuild.* modulefiles anymore, please delete all of them. Thanks!

lgannoaa replied:
Removed as requested.

aerorahul commented on modulefiles/OznMonBuild.jet:
```diff
@@ -0,0 +1,17 @@
+export ver=2.0.2
+export FCOMP=mpiifort
+
+export CF=$FCOMP
```
What is `CF`?

lgannoaa replied:
Please check with the original developer. I use the same approach:
grep "export CF" *
OznMonBuild.jet:export CF=$FCOMP
OznMonBuild.orion:export CF=$FCOMP
OznMonBuild.wcoss_dell_p3:export CF=$FCOMP
RadMonBuild.jet:export CF=$FCOMP
RadMonBuild.orion:export CF=$FCOMP
RadMonBuild.wcoss_dell_p3:export CF=$FCOMP

aerorahul replied:
Yeah, I don't think that is correct.
`CF` is not used in the GSI.  I checked.
These flags being set here are also ignored. 

I am sure you can eliminate lines 5-12 from all these files and it will be identical. 

Also, if the PR is adding functionality provided by some other developers, please invite them in the review.

KateFriedman-NOAA replied:
Same comment as above, please remove the modulefiles/OznMonBuild.* and modulefiles/RadMonBuild.* modulefiles. Thanks!

lgannoaa replied:
Removed as requested.

aerorahul commented on modulefiles/RadMonBuild.jet:
```diff
@@ -0,0 +1,18 @@
+# Modfule file for radmon
+########################################################
+export ver=2.0.2
+export FCOMP=mpiifort
+
+export CF=$FCOMP
+export FC=$FCOMP
```
same question as above.

lgannoaa replied:
As requested, I put the answer above.

KateFriedman-NOAA replied:
Same comment as above, please remove the modulefiles/OznMonBuild.* and modulefiles/RadMonBuild.* modulefiles. Thanks!

lgannoaa replied:
Removed as requested.

aerorahul commented on util/sorc/compile_gfs_util_jet.sh:
```diff
@@ -0,0 +1,57 @@
+
+if [ "$target" = "wcoss_dell_p3" ] || [ "$target" = "wcoss_cray" ] || [ "$target" = "jet" ] || [ "$target" = "hera" ] ; then
+   echo " "
+   echo " You are on WCOSS:  $target "
```
You are not on WCOSS if you are on Hera or Jet

lgannoaa replied:
Modified to display Hera and Jet

aerorahul commented on util/sorc/compile_gfs_util_jet.sh:
```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
```
What part of the workflow runs this on Jet?
Is this necessary for research purposes?
This seems like an operational product.

lgannoaa replied:
gfs_util is not been used in JET fcst only run. It is just a place holder here to complete the build without error. If we are sure JET will never need the gfs_util, I can remove it. However, the build script will need to be modified to filter out JET target.

aerorahul replied:
Please delete this file.
We will revisit this when there is a need.

lgannoaa replied:
Modified build_gfs_util.sh to filter out JET for now.

aerorahul replied:
This script is called `compile_gfs_util_jet.sh` and yet it has targets for `wcoss`, `wcoss_cray`, `hera` etc.

lgannoaa replied:
compile_gfs_util_jet.sh is deleted

aerorahul commented on modulefiles/workflow_utils.jet:
```diff
@@ -0,0 +1,32 @@
+module load g2/3.4.1
+module load bufr/11.4.0
+
+module load hdf5/1.10.6
+module load netcdf/4.7.4
```
you need to load these after png.
netcdf is used in w3 library.

lgannoaa replied:
I am testing this item on Jet now.

aerorahul commented on modulefiles/module_base.jet:
```diff
@@ -1,32 +1,27 @@
-module load mpiserial
-
+module load nco/4.9.1
+module load cdo/1.9.5
 module load gempak/7.4.2
-
+module load prod_util/v1.0.18
+module load grib_util/v1.1.1
+module load g2tmpl/1.6.0
+module load crtm/2.3.0
+module load esmflocal/8_0_1
+module load netcdfp/4.7.4
```
Please make sure these are built with hpc-stack.
I know for a fact hpc-stack does not build netcdfp or esmflocal modules for these

lgannoaa replied:
I am testing this item on Jet now.

aerorahul replied:
some of these are still not from hpc-stack.  So if you are testing with these modules, then they are likely wrong.

aerorahul reviewed (changes requested):
More comments.

KateFriedman-NOAA commented:
@lgannoaa The FV3 build fails when run out-of-the-box from your PR branch on Jet. Please see the following, thanks!

my clone of your branch: /lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only
log of FV3 build (as done via build_all.sh): /lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/sorc/logs/build_fv3.log

Also, the GLDAS and WW3 pre/post builds fail in an out-of-the-box build attempt:
* The GLDAS is understandable since it has not been ported to Jet yet. The full cycled/DA port to Jet is being taken care of by Dave Huber in issue #357.
* The WW3 pre/post build failed because a jet modulefile for it needs to be added (please remedy, thanks!):
```
 -bash-4.2$ tail /lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/sorc/logs/build_ww3_prepost.log
+ '[' '!' -d ../exec ']'
++ pwd -P
+ finalexecdir=/mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/sorc/../exec
+ set +x
./build_ww3prepost.sh: line 13: ../modulefiles/modulefile.ww3.jet: No such file or directory
```

aerorahul commented on sorc/build_gfs_util.sh:
```diff
@@ -18,4 +18,4 @@ echo ""
 echo ""
 
-source ./compile_gfs_util_wcoss.sh
+source ./compile_gfs_util_${target}.sh
```
please revert.

lgannoaa replied:
Removed compile_gfs_util_jet.sh and reverted build_gfs_util.sh

lgannoaa commented:
Hi @junwang-noaa,
The fv3gfs build failed on Jet because there is no ufs_jet in fv3gfs.fd/modulefiles. The script compile.sh is looking for ufs_jet but ufs_jet.intel and ufs_jet.intel_debug are available. Please advice solution.

aerorahul commented:
> Hi @junwang-noaa,
> The fv3gfs build failed on Jet because there is no ufs_jet in fv3gfs.fd/modulefiles. The script compile.sh is looking for ufs_jet but ufs_jet.intel and ufs_jet.intel_debug are available. Please advice solution.

@lgannoaa
You need to add in `build_fv3.sh`
```
if [ $target = jet ]; then target=jet.intel ; fi
```

The modulefile for jet is present in the hash that you are checking out.
Please take a minute to look and investigate.

```
~/s/fv3gfs.fd ❯❯❯ git branch
* (HEAD detached at 9350745)
/s/fv3gfs.fd ❯❯❯ ls modulefiles/ufs_jet.intel 
modulefiles/ufs_jet.intel
```

aerorahul commented:
> > Hi @junwang-noaa,
> > The fv3gfs build failed on Jet because there is no ufs_jet in fv3gfs.fd/modulefiles. The script compile.sh is looking for ufs_jet but ufs_jet.intel and ufs_jet.intel_debug are available. Please advice solution.
> 
> @lgannoaa
> You need to add in `build_fv3.sh`
> 
> ```
> if [ $target = jet ]; then target=jet.intel ; fi
> ```
> 
> The modulefile for jet is present in the hash that you are checking out.
> Please take a minute to look and investigate.
> 
> ```
> ~/s/fv3gfs.fd ❯❯❯ git branch
> * (HEAD detached at 9350745)
> /s/fv3gfs.fd ❯❯❯ ls modulefiles/ufs_jet.intel 
> modulefiles/ufs_jet.intel
> ```

@lgannoaa 
And if it doesn't build, then how was this tested prior to opening a PR?

lgannoaa commented:
Hi @DavidHuber-NOAA I created the following 6 module files to get gldas build working on Jet.
Please review them in JET:/mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow/sorc/gldas.fd/modulefiles
gldas_rst.jet gldas_post.jet gldas_model.jet gldas2gdas.jet gdas2gldas.jet gldas_forcing.jet
If you agree with this solution, please let us know and update gldas package as needed.
Thanks

lgannoaa commented:
> > > Hi @junwang-noaa,
> > > The fv3gfs build failed on Jet because there is no ufs_jet in fv3gfs.fd/modulefiles. The script compile.sh is looking for ufs_jet but ufs_jet.intel and ufs_jet.intel_debug are available. Please advice solution.
> > 
> > 
> > @lgannoaa
> > You need to add in `build_fv3.sh`
> > ```
> > if [ $target = jet ]; then target=jet.intel ; fi
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The modulefile for jet is present in the hash that you are checking out.
> > Please take a minute to look and investigate.
> > ```
> > ~/s/fv3gfs.fd ❯❯❯ git branch
> > * (HEAD detached at 9350745)
> > /s/fv3gfs.fd ❯❯❯ ls modulefiles/ufs_jet.intel 
> > modulefiles/ufs_jet.intel
> > ```
> 
> @lgannoaa
> And if it doesn't build, then how was this tested prior to opening a PR?

I used a local modified slink as a workaround on JET during testing. I will modify build_fv3.sh using your solution.

EdwardSafford-NOAA commented:
I can confirm that the modulefiles/RadMonBuild.jet and modulefiles/OznMonBuild.jet are no longer needed.  That change occurred when when we moved to building with cmake.

I'm afraid I can't offer much more; I don't have access to the jet machine.

lgannoaa commented:
> I can confirm that the modulefiles/RadMonBuild.jet and modulefiles/OznMonBuild.jet are no longer needed. That change occurred when when we moved to building with cmake.
> 
> I'm afraid I can't offer much more; I don't have access to the jet machine.

Thank you @EdwardSafford-NOAA for clarification. There are removed.

aerorahul commented:
> > > > Hi @junwang-noaa,
> 
> > > > The fv3gfs build failed on Jet because there is no ufs_jet in fv3gfs.fd/modulefiles. The script compile.sh is looking for ufs_jet but ufs_jet.intel and ufs_jet.intel_debug are available. Please advice solution.
> 
> > > 
> 
> > > 
> 
> > > @lgannoaa
> 
> > > You need to add in `build_fv3.sh`
> 
> > > ```
> 
> > > if [ $target = jet ]; then target=jet.intel ; fi
> 
> > > ```
> 
> > > 
> 
> > > 
> 
> > >     
> 
> > >       
> 
> > >     
> 
> > > 
> 
> > >       
> 
> > >     
> 
> > > 
> 
> > >     
> 
> > >   
> 
> > > The modulefile for jet is present in the hash that you are checking out.
> 
> > > Please take a minute to look and investigate.
> 
> > > ```
> 
> > > ~/s/fv3gfs.fd ❯❯❯ git branch
> 
> > > * (HEAD detached at 9350745)
> 
> > > /s/fv3gfs.fd ❯❯❯ ls modulefiles/ufs_jet.intel 
> 
> > > modulefiles/ufs_jet.intel
> 
> > > ```
> 
> > 
> 
> > @lgannoaa
> 
> > And if it doesn't build, then how was this tested prior to opening a PR?
> 
> 
> 
> I used a local modified slink as a workaround on JET during testing. I will modify build_fv3.sh using your solution.

So it was not tested entirely with this branch.  When someone opens a PR, it is expected that they tested it with the branch.

aerorahul commented:
> I can confirm that the modulefiles/RadMonBuild.jet and modulefiles/OznMonBuild.jet are no longer needed.  That change occurred when when we moved to building with cmake.
> 
> 
> 
> I'm afraid I can't offer much more; I don't have access to the jet machine.

Thank you @EdwardSafford-NOAA. Your confirmation was all that was required.

DavidHuber-NOAA commented:
> Hi @DavidHuber-NOAA I created the following 6 module files to get gldas build working on Jet.
> Please review them in JET:/mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow/sorc/gldas.fd/modulefiles
> gldas_rst.jet gldas_post.jet gldas_model.jet gldas2gdas.jet gdas2gldas.jet gldas_forcing.jet
> If you agree with this solution, please let us know and update gldas package as needed.
> Thanks

@lgannoaa Could you add read permissions for me to /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR?

lgannoaa commented:
> > Hi @DavidHuber-NOAA I created the following 6 module files to get gldas build working on Jet.
> > Please review them in JET:/mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow/sorc/gldas.fd/modulefiles
> > gldas_rst.jet gldas_post.jet gldas_model.jet gldas2gdas.jet gdas2gldas.jet gldas_forcing.jet
> > If you agree with this solution, please let us know and update gldas package as needed.
> > Thanks
> 
> @lgannoaa Could you add read permissions for me to /mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR?

Done.

KateFriedman-NOAA commented:
@lgannoaa @DavidHuber-NOAA GLDAS is unrelated to this task/PR (GLDAS is not needed for free-forecast mode), please move GLDAS Jet port discussions to the related GLDAS repo issue for the port. Thanks!

KateFriedman-NOAA commented:
@lgannoaa Friendly reminder to follow the global-workflow commit message standards, see here:

https://github.com/NOAA-EMC/global-workflow/wiki/Development-of-Global-Workflow#commit-message-standards

Thanks!

lgannoaa commented:
I modified the build_gldas.sh, build_ww3prepost.sh, and build_gfs_util.sh to not build on Jet.

KateFriedman-NOAA commented:
> I modified the build_gldas.sh, build_ww3prepost.sh, and build_gfs_util.sh to not build on Jet.

@lgannoaa Please do not disable those builds. The ww3prepost build just needs the modulefile added (see my comment above). The gfs_util and gldas builds will be taken care of separately. Thanks!

lgannoaa commented:
> @lgannoaa Friendly reminder to follow the global-workflow commit message standards, see here:
> 
> https://github.com/NOAA-EMC/global-workflow/wiki/Development-of-Global-Workflow#commit-message-standards
> 
> Thanks!

@KateFriedman-NOAA you reported the GLDAS not build in the first place. Maybe you can also include a request for relocate that discussion in the same message instead of revisit the topic later.

lgannoaa commented:
> > I modified the build_gldas.sh, build_ww3prepost.sh, and build_gfs_util.sh to not build on Jet.
> 
> @lgannoaa Please do not disable those builds. The ww3prepost build just needs the modulefile added (see my comment above). The gfs_util and gldas builds will be taken care of separately. Thanks!

@KateFriedman-NOAA if individual build script is not disabled, the overall build process will have fail status. That may introduce confusion. I recommend to disable it (it is currently failed) for now and enable it when the build is actually working.

lgannoaa committed:
Modified to remove OznMon RadMon modulefiles
Disable JET build for unsupported package gfs_util, gldas, and ww3
Update python scripts in ush/rocoto
	deleted:    modulefiles/OznMonBuild.jet
	deleted:    modulefiles/RadMonBuild.jet
	modified:   sorc/build_fv3.sh
	modified:   sorc/build_gfs_util.sh
	modified:   sorc/build_gldas.sh
	modified:   sorc/build_ww3prepost.sh
	modified:   ush/rocoto/rocoto.py
	modified:   ush/rocoto/setup_expt_fcstonly.py
	modified:   ush/rocoto/setup_workflow_fcstonly.py
	modified:   ush/rocoto/workflow_utils.py
	deleted:    util/sorc/compile_gfs_util_jet.sh

lgannoaa commented:
@aerorahul I tested this package on JET under:
/mnt/lfs4/HFIP/hfv3gfs/Lin.Gan/PR/global-workflow
It build successfully.
It created expdir and xml file.

aerorahul commented on sorc/build_gfs_util.sh:
```diff
@@ -14,6 +14,12 @@ if [ ! -d "./mkgfsawps.fd" ]; then
    exit
 fi 
 
+if [ $target = jet ]; then
+   echo "  GFS_UTIL does not support JET "
+   echo "  "
+   exit
+fi
+
```
don't need this block.  please remove.
We don't check for every machine here, so no need to do check for jet here.

aerorahul commented on sorc/build_gldas.sh:
```diff
@@ -4,6 +4,12 @@ set -eux
 source ./machine-setup.sh > /dev/null 2>&1
 cwd=`pwd`
 
+if [ $target = jet ]; then
+   echo "  GLDAS does not support JET "
+   echo "  "
+   exit
+fi
+
```
Please remove.

aerorahul commented on sorc/build_ww3prepost.sh:
```diff
@@ -10,6 +10,13 @@ finalexecdir=$( pwd -P )/../exec
 
 set +x
 source ./machine-setup.sh > /dev/null 2>&1
+
+if [ $target = jet ]; then
+   echo "  WW3 does not support JET "
+   echo "  "
+   exit
+fi
+
```
please remove

aerorahul commented on util/modulefiles/gfs_util.jet:
```diff
@@ -0,0 +1,31 @@
+#%Module#####################################################
```
we are not building this, so please remove.

aerorahul commented:
@lgannoaa 
Your branch is now out of date with `develop` which is adding files that are in direct conflict.
There are too many unresolved comments in this PR that is making it impossible to track what is meaningful v/s unnecessary or irrelevant.

aerorahul commented on ush/rocoto/rocoto.py:
```diff
@@ -33,17 +31,17 @@ def create_metatask(task_dict, metatask_dict):
     strings = []
 
-    strings.append(f'<metatask name="{metataskname}">\n')
+    strings.append('<metatask name="%s">\n' % metataskname)
```
This file is an older version. Please merge develop into your branch.

aerorahul reviewed (changes requested):
This PR is not even close for a review, let alone testing and merge to develop.
New files are added over every iteration of the review.
The PR description lists 6 files, and yet there are 13 files with changes.

lgannoaa commented:
> This PR is not even close for a review, let alone testing and merge to develop.
> New files are added over every iteration of the review.
> The PR description lists 6 files, and yet there are 13 files with changes.

@aerorahul We have communicated all day on this PR. I worked with all your requests and changed my approach. It is normal to have files changed from this kind of activity. I am surprised to see this comment above.

Here is the justification for file changes:
- OznMonBuild.jet and RadMonBuild.jet are the result of a group discussion today. Ed told us this can be removed.
- build_fv3.sh is the result of a group discussion today. The module name is incorrect in JET fv3 build. You proposed a solution and I used it. 
- build_gfs_util.sh, build_gldas.sh, build_ww3prepost.sh also discussed today where it does not build on JET. Therefore it should be disabled on JET to allow the build process to complete.
- The current global-workflow version of setup_expt_fcstonly.py, setup_workflow_fcstonly.py,  rocoto.py, and workflow_utils.py does not work on JET. There are syntax errors. I had to revert this auto-merged version back to the version I used before.

I recommend having a clear and complete request. For example,
1. The hpc-stack is used in module_base.jet because of your request today. It was built without an error. There is no communication today on why it is wrong or what should be changed.
2. If a package does not need to be built on JET, it should be clearly stated within the request to either disable the build script within the package or modify the main build all script. We can not just let it build and fail on JET.

@arunchawla-NOAA I worked with physics group to test my JET solution and they have accepted and are using it. I understand people may have different opinions on how things can be done. However, I do not understand why we see the statement this PR is not even close for a review.

aerorahul commented:
Because we are in the middle of a discussion. There are many issues to be resolved before it is ready for merge to develop. 

GitHub provides a means to facilitate discussion by creating a Draft Pull Request. When the discussion is concluded, there is a button to mark this PR for Review. 

If there are issues with syntax in the rocoto Python scripts, this is the first we heard about it in the PR. Please open a new issue for it as if these are syntax issues, they need to be handled urgently and separately. 

The build for fv3 were clearly not correct as you were using a version built offline, probably with different modules than those in this PR. We caught that in this discussion. 

Jet uses hpc-stack and we test the model routinely on jet with it. The modules you were loading are something else and unsupported. We caught that in this discussion. 

The changes to disabling components needs to be handled in a better way than scattering if blocks across numerous scripts. That is not an opinion, it is practical programming for the future. 

I am happy to finish the discussion offline. 

The hash that this PR was opened on did not build on Jet. That in itself is a red flag.  You admitted that you used a model executable built elsewhere to test this branch. That circumvents the build script in this repository. 

I am not sure what the physics group tested for you, but I am fairly certain they did not checkout this branch, inspect how the model is being built and cleanly build and run their tests. Because if they had, they would have failed at the model.

lgannoaa commented:
Close this PR as requested and will open a draft next week.
Will start communicating offline.

BTW, physics group used my local copy of HOMEgfs on JET. This approach was the part of the JET task requirement assigned by Arun.
